synapse/handlers/space_summary.py

```diff
@@ -420,9 +421,8 @@ async def _is_room_accessible(
 
         It should be included if:
 
-        * The requester is joined or invited to the room.
-        * The requester can join without an invite (per MSC3083).
-        * The origin server has any user that is joined or invited to the room.
+        * The requester is joined or can join the room (per MSC3173).
+        * The origin server has any user that is joined or can join the room.
         * The history visibility is set to world readable.
```

Comment:
I debated making all of this a method on `Auth`: `can_user_view_stripped_state` or something, but seemed a bit silly until we want to use it in another place.